### ISSUE TITLE
Extension of generic module fails reflexive equality when upcast

### ISSUE THREAD

**Issue by ezrast** [crystal-lang/crystal#6760]
Splitting from #2665 at @RX14's request:
```crystal
module Base(T)
end

module Derived
  extend Base(Int32)
end

arr = [Derived] of Base(Int32)
p! arr.first == arr.first # false; should be true
p! Derived == arr.first   # false; should be true
p! arr.first == Derived   # true
```
I'm not convinced that this shouldn't compile; it makes sense to me that one module extending another establishes an `is_a?` relationship between them (although "derived" may not be the correct term to use here).

Carc: https://carc.in/#/r/5239
Version: Crystal 0.26.1

**Comment by ezrast** [comment]
I'm extending `Base(Int32)`, not including it. An instance of `Derived` would definitely not be a `Base(Int32)`.

Are you saying this behavior is not intended?
```crystal
module A
end

module B
  extend A
end

p B.is_a? A  # true
p {{ B.class.ancestors.includes? A }}  # true
```

**Comment by HertzDevil** [comment]
Seems to be fixed in 1.1.0?

**Comment by asterite** [comment]
Yup, fixed, thanks!
